src/patternfly/components/TextInputGroup/text-input-group.scss

```diff
@@ -35,12 +38,28 @@
   --pf-c-text-input-group__utilities--c-button--PaddingRight: var(--pf-global--spacer--xs);
   --pf-c-text-input-group__utilities--c-button--PaddingLeft: var(--pf-global--spacer--xs);
 
+  // input disabled style
+  --pf-c-text-input-group--m-disabled--Color: var(--pf-global--disabled-color--100);
+  --pf-c-text-input-group--m-disabled--BackgroundColor: var(--pf-global--disabled-color--300);
+  --pf-c-text-input-group--m-disabled--BorderColor: transparent;
+
   position: relative;
   display: flex;
+  color: var(--pf-c-text-input-group--Color);
+  background-color: var(--pf-c-text-input-group--BackgroundColor);
 
   &:hover {
     --pf-c-text-input-group__text--after--BorderBottomColor: var(--pf-c-text-input-group--hover__text--after--BorderBottomColor);
   }
+
+  &.pf-m-disabled {
+    --pf-c-text-input-group--Color: var(--pf-c-text-input-group--m-disabled--Color);
+    --pf-c-text-input-group__icon--Color: var(--pf-c-text-input-group--m-disabled--Color);
+    --pf-c-text-input-group--BackgroundColor: var(--pf-c-text-input-group--m-disabled--BackgroundColor);
+
+    cursor: not-allowed;
```

Comment:
typically we will use `pointer-events: none;` here, which will disable any kind of interaction within the component.